src/components/ProfileCard/GiftInput.tsx

```diff
@@ -0,0 +1,160 @@
+import { transparentize } from 'polished';
+
+import { IconButton, makeStyles } from '@material-ui/core';
+
+import { ApeTextField } from 'components';
+import { PlusCircleIcon, MinusCircleIcon } from 'icons';
+import { useSelectedCircle } from 'recoilState';
+
+import { CardInfoText } from './CardInfoText';
+
+import { TUpdateGift } from 'types';
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    display: 'flex',
+    flexDirection: 'column',
+    width: '100%',
+    padding: theme.spacing(0, 0.5),
+  },
+  textField: {
+    marginTop: theme.spacing(4),
+    '& label': {
+      height: 14,
+      margin: theme.spacing(0, 0, 0.5),
+      fontSize: 12,
+      fontWeight: 'bold',
+      color: 'rgba(81, 99, 105, 0.7)',
+    },
+  },
+  tokenInputContainer: {
+    height: 66,
+    marginTop: theme.spacing(1.5),
+    marginBottom: 0,
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'end',
+  },
+  noteTextarea: {
+    '& textarea': {
+      fontSize: 12,
+      wordBreak: 'normal',
+    },
+    '&.MuiInputBase-root': {
+      backgroundColor: transparentize(0.3, theme.colors.white),
+      padding: theme.spacing(1.5, 0),
+    },
+  },
+  spinButton: {
+    '&:first-of-type': {
+      paddingLeft: theme.spacing(1),
+      paddingRight: 0,
+    },
+    '&:last-of-type': {
+      paddingLeft: 0,
+      paddingRight: theme.spacing(1),
+    },
+  },
+  tokenInput: {
+    '& input': {
+      fontSize: 25,
+    },
+    '&.MuiInputBase-root': {
+      backgroundColor: 'white',
+    },
+    width: 140,
+    height: 36,
+  },
+}));
+
+export const GiftInput = ({
+  tokens,
+  note,
+  updateGift,
+}: {
+  note: string;
+  tokens?: number;
+  updateGift: TUpdateGift;
+}) => {
+  const classes = useStyles();
+  const selectedCircle = useSelectedCircle();
+
+  const onChangeTokens = ({
+    target: { value },
+  }: React.ChangeEvent<HTMLInputElement>) => {
+    try {
+      updateGift({ tokens: Number(value) });
+    } catch (e) {
+      console.warn('onChangeTokens: Not a number.');
+    }
+  };
+
+  const onChangeNote = ({
+    target: { value },
+  }: React.ChangeEvent<HTMLTextAreaElement>) => updateGift({ note: value });
+
+  return (
+    <div className={classes.root}>
+      {tokens !== undefined ? (
+        <ApeTextField
+          label={`${selectedCircle?.tokenName} Allocated`}
```

Comment:
it's weird to me to see strings like this because that implies it's expected behavior to render "undefined Allocated" in some circumstances

Reply:
Maybe the whole component should short-circuit and render nothing if !selectedCircle.

Or perhaps components like this should depend upon normal props for their circles rather than using a hook. 

This is the sort of thing that I was getting at when I said earlier today that hooks are a cursed abstraction. Functional components are powerful and composable for the same reason that functional programming in general is powerful and composable: with a pure function, you know that its output depends only on its input, not any non-local state. So it's easy to reason about and control, because the scope of things you have to consider in order to understand what a piece of code will do is clearly defined by the code itself. But hooks allow functional components to depend on non-local state, so this benefit goes out the window. \</soapbox>

Reply:
I think normal props make sense for "dumb" components.

I'm imagining you long ago `<soapbox>`
Good point.